Automattic/jetpack
# Fixes the response body for the errors returned by the Blaze module

The blaze module is overriding the data that the DSP server is returning in the errors, with a body like this:
```
{"code":"remote-error","message":"unknown remote error","data":{"status":500}}
```

The Blaze create campaign widget has some specific logic that runs on certain errors, and we can't lose the body that the DSP API gives us. 

This PR attempts to maintain the code and status in the error response to make it compatible with other WPCOM controllers but also returns the body that the Widget expects.

The final response is something like this:
```
{
    "errorMessage": "Error fetching WPcom payment methods",
    "invalidToken": true,
    "code": "remote-error",
    "status": 500
}
```

## Proposed changes:

* Change the error responses of the Blaze module to be more compatible with the Blaze create campaign widget

### Other information:


## Jetpack product discussion


N/A

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






We will need to generate an error in the DSP API. One way to do it is by invalidating the user token in our DB. Ping me, and I will help you test this. If you have already setup a local DSP installation, you can sandbox the DSP API and clear you local social login token.

* Go to Tools->Advertising
* Click on the Promote button for any of your posts
* Verify that the errors returned by the Blaze module follow the example shown before
* If you cleared up your DSP Token, verify that the post create campaign widget finish loading the page with valid data
